src/checkers/inference/model/serialization/ToStringSerializer.java

```diff
@@ -233,34 +263,27 @@ public String serialize(ExistentialVariableSlot slot) {
     public String serialize(CombVariableSlot slot) {
         final StringBuilder sb = new StringBuilder();
         sb.append(slot.getId());
-
         if (showVerboseVars) {
-            sb.append(": merges ");
-            sb.append(Arrays.asList(slot.getFirst(), slot.getSecond()));
+            sb.append(": merges ")
+              .append(Arrays.asList(slot.getFirst(), slot.getSecond()));
             formatMerges(slot, sb);
             optionallyFormatAstPath(slot, sb);
         }
         return sb.toString();
     }
 
     protected String indent(String str) {
-        if (indent == 0) {
-            return str;
-        }
-
-        StringBuilder sb = new StringBuilder();
-        for (int i = 0; i < indent; i++) {
-            sb.append(INDENT_STRING);
-        }
-
-        sb.append(str);
+        generateIndentations();
```

Comment:
It seems `generateIndentations()` try to encapsulate the generation logic from this `indent()` method.

However, since `indent()` method is the only client use it, why encapsulate it?

It makes code a bit harder to read as now I need to jump through methods...

It would be more clear to just embed the catching loop within the `indent` method, just like other lazy-evaluation method, we write:
```java
getObj() {
if (obj == null) new Object();
return obj;
}
```

But not:
```java
getObj() {
 if (obj == null) createNewObject();
return obj;
}

//somewhere else
void createNewObject() {
  obj = new Object();
}
```